### ISSUE TITLE
TS Errors when importing ReactLinearGenomeView

### ISSUE THREAD

**Issue by joneubank** [GMOD/jbrowse-components#3765]
**Describe the bug**

Projects importing [JBrowse React Linear View](https://www.npmjs.com/package/@jbrowse/react-linear-genome-view) have errors running `tsc`. There are `*.ts` in the `/src` directory of the imported dependency, inside `node_modules/`, and these are failing TS type checking.

**To Reproduce**

A repository with a minimal project showing how to reproduce is here: https://github.com/joneubank/minimal-reproduce-jbrowse-component-build

Steps to reproduce and a further description of the errors are in the README of that repo.

**Expected behavior**

Should be able to run `tsc` build on a project importing JBrowse React Linear Genome View, without having to modify our `tsconfig` or needing additional 3rd party dependencies.

**Screenshots**

**Version:**

```
"dependencies": {
    "@jbrowse/react-linear-genome-view": "^2.6.1",
    "react": "18.2.0",
    "react-dom": "18.2.0"
  },
  "devDependencies": {
    "@types/react": "18.0.25",
    "@types/react-dom": "18.0.9",
    "typescript": "5.1.3"
  },
  ```

**Additional context**

This issue did not occur for us with the Circular Genome View, even though that repo also exports the `/src` folder. I would recommend removing the `/src` files from the exported pacakges to prevent this kind of error in the future.

This relates to https://github.com/GMOD/jbrowse-components/issues/3764 and the two may in fact be duplicates now.

**Comment by cmdcolin** [comment]
that is really odd that adding @types/jest fixes all the errors. I think I can confirm locally too.

we can likely do the inlinesourcemaps change and see if that helps though

**Comment by joneubank** [comment]
Yeah the jest import thing confounds me... `inlinesourcemaps` seems like a promising route though, happy to trial any test builds you can provide.

**Comment by cmdcolin** [comment]
@joneubank published v2.6.2 with no source maps/src folder (inline source maps still needed src folder to do properly, but we dont mangle our builds too much so just published with no source map)

also fixed an error with the @jbrowse/react-circular-genome-view. lemme know if there are any other issues you run into!

**Comment by cmdcolin** [comment]
(the react-circular-genome-view issue that was fixed was properly publishing commonjs code in the dist folder)
